src/backend/tools/lang_chain.py

```diff
@@ -31,13 +31,14 @@ def __init__(self, chunk_size: int = 300, chunk_overlap: int = 0):
     def is_available(cls) -> bool:
         return True
 
-    def retrieve_documents(self, query: str, **kwargs: Any) -> List[Dict[str, Any]]:
+    def call(self, parameters: str, **kwargs: Any) -> List[Dict[str, Any]]:
         wiki_retriever = WikipediaRetriever()
-        docs = wiki_retriever.get_relevant_documents(query)
+        docs = wiki_retriever.get_relevant_documents(parameters)
```

Comment:
We need to extract "query" from parameters

Reply:
Done, modified so I pass in a query param